traits/trait_types.py

```diff
@@ -1029,14 +1029,6 @@ class Constant(TraitType):
     #: The standard metadata for the trait:
     metadata = {"type": "constant", "transient": True}
 
-    def __init__(self, value, **metadata):
-        if type(value) in MutableTypes:
```

Comment:
I think `MutableTypes` is not used by anything else, so can probably be removed completely.

Reply:
Removed.